labs-ios-starter/Surveys/Model Controllers/TopicController.swift

```diff
@@ -184,26 +184,26 @@ class TopicController {
     }
 
     // MARK: - Delete
-        /// Deletes a topic from the server
-        /// - Warning: This method should `ONLY` be called by the leader of a topic for their own survey
-        func deleteTopic(topic: Topic, completion: @escaping CompleteWithNetworkError) {
-            guard
-                let id = topic.id,
-                let deleteRequest = createRequest(pathFromBaseURL: "topic/\(id)", method: .delete)
-            else {
-                completion(.failure(.badRequest))
-                return
-            }
+    /// Deletes a topic from the server
+    /// - Warning: This method should `ONLY` be called by the leader of a topic for their own survey
```

Comment:
Could we use a predicate here to ensure that only the leader is deleting the topic?